sys/console/full/src/console.c

```diff
@@ -116,7 +116,11 @@ console_read(char *str, int cnt, int *newline)
     }
     cmd = ev->ev_arg;
     len = strlen(cmd->line);
-    strncpy(str, cmd->line, len+1);
+    if (cnt < len) {
+        len = cnt;
```

Comment:
We have to fix the buffer overrun one way or another :)
If I remember correctly, before new console code, it was copying up to cnt - 1 characters, and null terminating
the string. My opinion would be to keep it like that, and update documentation like you suggested.